catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/NewDataDialogTest.java

```diff
@@ -0,0 +1,222 @@
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testInvalidVarNames() {
+		String emptyVarName = "";
+		String spaceVarName = " ";
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(emptyVarName);
+
+		clickOk();
+
+		checkErrorMessageDisplayed(R.string.name_consists_of_spaces_only);
+
+		onView(withId(R.id.input_edit_text))
+				.perform(typeText(spaceVarName));
+
+		clickOk();
+
+		checkErrorMessageDisplayed(R.string.name_consists_of_spaces_only);
+
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testInvalidThenValidVarNameErrorMessage() {
+		String spaceVarName = " ";
+		String validVarName = "name";
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(spaceVarName);
+
+		checkErrorMessageDisplayed(R.string.name_consists_of_spaces_only);
+
+		onView(withId(R.id.input_edit_text))
+				.perform(typeText(validVarName));
+
+		clickOk();
+
+		checkVariableShownInSpinner(validVarName);
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testValidSpecialCharVarNames() {
+		String underscoreVarName = "var_var";
+		String numberVarName = "var123";
+		String validSpecialCharVarName = "var$"; // § doesn't exist in android?
+		String slashVarName = "var/";
+		String bracketVarName = "var(";
+		String curlyBracketVarName = "var{";
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(underscoreVarName);
+		checkVariableShownInSpinner(underscoreVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(numberVarName);
+		checkVariableShownInSpinner(numberVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(validSpecialCharVarName);
+		checkVariableShownInSpinner(validSpecialCharVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(slashVarName);
+		checkVariableShownInSpinner(slashVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(bracketVarName);
+		checkVariableShownInSpinner(bracketVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(curlyBracketVarName);
+		checkVariableShownInSpinner(curlyBracketVarName);
+
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testCancelVarCreation() {
+		openNewDataDialog(variableBrickPosition, R.id.set_variable_spinner);
+
+		clickCancel();
+
+		onBrickAtPosition(variableBrickPosition)
+				.checkShowsText(R.string.brick_variable_spinner_create_new_variable);
+	}
+
+	private void openNewDataDialog(int brickPosition, int spinnerId) {
```

Comment:
this is kind of misleading for the next guy working on this class, since this only opens a new data dialog if theres no variable selected or set before.